oauth2.go

```diff
@@ -208,3 +216,13 @@ func (tv *tokenValidator) validate(token, tokenTypeHint string) (*introspectionR
 	}
 	return response, nil
 }
+
+// parseToken takes a string and unmarshals to an oauth2 Token
+func parseToken(stringToken string) (*oauth2.Token, error) {
```

Comment:
nitpick: I think including the type `string` in the name `stringToken` is redundant here. Could just be `token` or `t` or `data` or something.

Reply:
yup, good point. I will omit down to `t`